astrodata/tests/test_nddata.py

```diff
@@ -1,58 +1,32 @@
 #!/usr/bin/env python
-import warnings
 
 import numpy as np
 import pytest
 
 from astrodata import nddata
 
 
-def test_new_variance_uncertainty_instance_returns_none():
-    result = nddata.new_variance_uncertainty_instance(None)
+def test_variance_uncertainty_warn_if_there_are_any_negative_numbers():
+    arr = np.zeros((5, 5))
+    arr[2, 2] = -0.001
 
-    assert result is None
+    with pytest.warns(RuntimeWarning, match='Negative variance values found.'):
+        result = nddata.ADVarianceUncertainty(arr)
 
+    assert not np.all(arr >= 0)
+    assert isinstance(result, nddata.ADVarianceUncertainty)
+    assert result.array[2, 2] == 0
 
-def test_new_variance_uncertainty_instance_no_warning_if_the_array_is_zeros():
-    array = np.zeros((500, 500), dtype=np.float32)
+    # check that it always works with a VarianceUncertainty instance
+    result.array[2, 2] = -0.001
 
-    with warnings.catch_warnings(record=True) as w:
-        result = nddata.new_variance_uncertainty_instance(array)
-        warnings.simplefilter("always")
+    with pytest.warns(RuntimeWarning, match='Negative variance values found.'):
+        result2 = nddata.ADVarianceUncertainty(result)
 
-        for _w in w:
-            print(str(_w.message))
-
-        assert (array >= 0).all()
-        assert len(w) == 0
-
-
-def test_new_variance_uncertainty_instance_warn_if_there_are_any_negative_numbers():
-    array = np.zeros((500, 500), dtype=np.float32)
-    array[250, 250] = -0.001
-
-    with warnings.catch_warnings(record=True) as w:
-        result = nddata.new_variance_uncertainty_instance(array)
-        warnings.simplefilter("always")
-
-        for _w in w:
-            print(str(_w.message))
-
-        assert not (array >= 0).all()
-        assert len(w) == 1
-        assert isinstance(result, nddata.StdDevUncertainty)
-        assert result.array[250, 250] == 0
-
-
-def test_new_variance_uncertainty_instance():
-    array = np.ones((500, 500), dtype=np.float32)
-
-    with warnings.catch_warnings(record=True) as w:
-        result = nddata.new_variance_uncertainty_instance(array)
-        warnings.simplefilter("always")
-
-        assert len(w) == 0
-        assert isinstance(result, nddata.StdDevUncertainty)
+    assert not np.all(arr >= 0)
+    assert not np.all(result.array >= 0)
```

Comment:
Yes, we check that the inputs are not modified, and the repetition is because in the first case the input is an `ndarray`, and in the second one it is an `ADVarianceUncertainty` instance.